src/az390.java

```diff
@@ -10923,6 +10954,196 @@ private void fp_get_hex(){
+				break;
+			case 8: // tz390.fp_lh_type s1,e7,m112 with split hex	
+				dc_hex = "EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE";
+				break;
+			case 9: // tz390.fp_lq_type quad word RPI 1108
+				dc_hex = "00000000000000000000000000000000";
+				break;	
+			}
+			if (fp_sign == '-'){
+				dc_hex = "8" + dc_hex.substring(1);
+			}
+			return;
+			
+		} else if (fp_text.toUpperCase().equals("(SNAN)")){ // #423
+			switch (tz390.fp_type){  // gen (min) hex for tz390.fp_type
+			case 0: // tz390.fp_db_type s1,e11,m52 with assumed 1
+				dc_hex = "7FFF7FFFFFFFFFFF";
+			    break;
+			case 1: // tz390.fp_dd_type s1,cf5,bxcf8,ccf50
+				dc_hex = "EEEEEEEEEEEEEEEE";
+				break;
+			case 2: // tz390.fp_dh_type s1,e7,m56 with hex exp
+				dc_hex = "EEEEEEEEEEEEEEEE";
+				break;
+			case 3: // tz390.fp_eb_type s1,e7,m24 with assumed 1
+	            dc_hex = "7F7FFFFF";
+	            break;
+			case 4: // tz390.fp_dd_type s1,cf5,bxcf6,ccf20
+				dc_hex = "EEEEEEEE"; // RPI 407
+				break;
+			case 5: // tz390.fp_eh_type s1,e7,m24 with hex exp
+				dc_hex = "EEEEEEEE";
+				break;
+			case 6: // tz390.fp_lb_type s1,e15,m112 with assumed 1
+				dc_hex = "7FFF7FFFFFFFFFFFFFFFFFFFFFFFFFFF";
+				break;
+			case 7: // tz390.fp_ld_type s1,cf5,bxcf12,ccf110
+				dc_hex = "EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"; // RPI 407
+				break;
+			case 8: // tz390.fp_lh_type s1,e7,m112 with split hex	
+				dc_hex = "EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE";
+				break;
+			case 9: // tz390.fp_lq_type quad word RPI 1108
+				dc_hex = "00000000000000000000000000000000";
+				break;	
+			}
+			if (fp_sign == '-'){
+				dc_hex = "8" + dc_hex.substring(1);
+			}
+			return;			
+
+		} else if (fp_text.toUpperCase().equals("(DMIN)")){
+			switch (tz390.fp_type){  // gen (DMIN) hex for tz390.fp_type
+			case 0: // tz390.fp_db_type s1,e11,m52 with assumed 1
+				dc_hex = "0000000000000001";
+			    break;
+			case 1: // tz390.fp_dd_type s1,cf5,bxcf8,ccf50
+				dc_hex = "0000000000000001"; // RPI 407
+				break;
+			case 2: // tz390.fp_dh_type s1,e7,m56 with hex exp
+				dc_hex = "0000000000000001";
+				break;
+			case 3: // tz390.fp_eb_type s1,e7,m24 with assumed 1
+	            dc_hex = "00000001";
+	            break;
+			case 4: // tz390.fp_dd_type s1,cf5,bxcf6,ccf20
+				dc_hex = "00000001"; // RPI 407
+				break;
+			case 5: // tz390.fp_eh_type s1,e7,m24 with hex exp
+				dc_hex = "00000001";
+				break;
+			case 6: // tz390.fp_lb_type s1,e15,m112 with assumed 1
+				dc_hex = "00000000000000000000000000000001";
+				break;
+			case 7: // tz390.fp_ld_type s1,cf5,bxcf12,ccf110
+				dc_hex = "00000000000000000000000000000001"; // RPI 407
+				break;
+			case 8: // tz390.fp_lh_type s1,e7,m112 with split hex	
+				dc_hex = "00000000000000007200000000000001";
+				break;
+			case 9: // tz390.fp_lq_type quad word RPI 1108
+				dc_hex = "00000000000000000000000000000000";
+				break;	
+			}
+			if (fp_sign == '-'){
+				dc_hex = "8" + dc_hex.substring(1);
```

Comment:
In this block, please replace all the literal magic numbers by named constants. And please use names that match the naming conventions used in the PoP

Reply:
These constants don't have any names that I can find.

Reply:
we can assign them. I'm much in favor of following IBM terminology. But if not provided, we're free to design our own in z390. I suggest [pos|neg]_[minval|maxval|infinity]_[db|dh|ed|eh|lb|ld|lh|lq]
Please modify or extend as needed...